spark/src/main/java/org/opensearch/sql/spark/dispatcher/SparkQueryDispatcher.java

```diff
@@ -117,6 +123,17 @@ public String cancelJob(AsyncQueryJobMetadata asyncQueryJobMetadata) {
           new InteractiveQueryHandler(sessionManager, jobExecutionResponseReader, leaseManager);
     } else if (IndexDMLHandler.isIndexDMLQuery(asyncQueryJobMetadata.getJobId())) {
       queryHandler = createIndexDMLHandler(emrServerlessClient);
+    } else if (asyncQueryJobMetadata.getJobType() == JobType.BATCH) {
+      queryHandler =
+          new RefreshQueryHandler(
```

Comment:
JobType.BATCH always means it's a refresh job? When is the `BatchQueryHandler` on L139 used?

Reply:
Yes, always refresh Job.
BatchQueryHandler it is used by UT only. we can refactor this part.